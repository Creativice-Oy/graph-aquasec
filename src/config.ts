import {
  IntegrationExecutionContext,
  IntegrationInstanceConfigFieldMap,
  IntegrationInstanceConfig,
} from '@jupiterone/integration-sdk-core';

/**
 * A type describing the configuration fields required to execute the
 * integration for a specific account in the data provider.
 *
 * When executing the integration in a development environment, these values may
 * be provided in a `.env` file with environment variables. For example:
 *
 * - `CLIENT_ID=123` becomes `instance.config.clientId = '123'`
 * - `CLIENT_SECRET=abc` becomes `instance.config.clientSecret = 'abc'`
 *
 * Environment variables are NOT used when the integration is executing in a
 * managed environment. For example, in JupiterOne, users configure
 * `instance.config` in a UI.
 */
export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
  /**
   * NOTE: Aqua calls this an "API Key", but it is essentially an auto-generated
   * API key _name_. The value of this property is _visible_ from the Aqua
   * console. The actual sensitive token is the `apiSecret` described below.
   */
  apiKey: {
    type: 'string',
  },
  apiSecret: {
    type: 'string',
    mask: true,
  },
  accountId: {
    type: 'string',
  },
};

/**
 * Properties provided by the `IntegrationInstance.config`. This reflects the
 * same properties defined by `instanceConfigFields`.
 */
export interface IntegrationConfig extends IntegrationInstanceConfig {
  apiKey: string;
  apiSecret: string;
  accountId: string;
}

export async function validateInvocation(
  context: IntegrationExecutionContext<IntegrationInstanceConfig>,
) {
  await Promise.resolve();
}
