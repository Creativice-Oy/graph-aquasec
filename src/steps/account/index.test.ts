import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';

import { IntegrationConfig } from '../../config';
import { fetchAccountDetails } from './index';
import { integrationConfig, withRecording } from '../../../test/config';

describe('#fetchAccount', () => {
  test(
    'should collect data',
    withRecording('fetchAccount', __dirname, async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: integrationConfig,
      });

      await fetchAccountDetails(context);

      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships:
          context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();

      const accounts = context.jobState.collectedEntities.filter((e) =>
        e._class.includes('Account'),
      );

      expect(accounts.length).toBeGreaterThan(0);
      expect(accounts).toMatchGraphObjectSchema({
        _class: ['Account'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'aquasec_account' },
            _rawData: {
              type: 'array',
              items: { type: 'object' },
            },
            manager: { type: 'string' },
            additionalCredits: { type: 'number' },
            associateTags: { type: 'boolean' },
            awsMarketplaceId: { type: 'string' },
            blockDefaultGroup: { type: 'boolean' },
            createdOn: { type: 'number' },
            cspEnabled: { type: 'boolean' },
            currentPlan: { type: 'number' },
            disableDefaultEmails: { type: 'boolean' },
            displayName: { type: 'string' },
            enforceOauth: { type: 'boolean' },
            globalSuppressionsAdminOnly: { type: 'boolean' },
            id: { type: 'string' },
            integrationsAdminOnly: { type: 'boolean' },
            name: { type: 'string' },
            numAllowedKeys: { type: 'number' },
            presuppressPlugins: { type: 'boolean' },
            rootUser: { type: 'string' },
            source: { type: 'string' },
            stripeId: { type: 'string' },
            suppressionsAdminOnly: { type: 'boolean' },
            trialEnd: { type: 'number' },
            usersAdminOnly: { type: 'boolean' },
          },
        },
      });
    }),
  );
});
