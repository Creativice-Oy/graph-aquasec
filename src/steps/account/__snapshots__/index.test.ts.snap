// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAccount should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "aquasec_account:10630",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "additional_credits": 0,
            "associate_tags": false,
            "aws_marketplace_id": "",
            "block_default_group": false,
            "byok_bucket": null,
            "created": "2021-10-09T00:36:24.000Z",
            "csp_enabled": true,
            "csp_metadata": null,
            "current_plan": 57,
            "custom_cost": null,
            "disable_default_emails": false,
            "enforce_oauth": false,
            "global_suppressions_admin_only": false,
            "id": 10630,
            "integrations_admin_only": false,
            "just_in_time_domain": null,
            "just_in_time_domain_secondary": null,
            "login_timeout": null,
            "num_allowed_keys": 1,
            "presuppress_plugins": false,
            "renewal_date": null,
            "restrict_to_ips": null,
            "root_user": "austin.kelleher@jupiterone.dev",
            "source": "aquasec",
            "stripe_id": null,
            "suppressions_admin_only": false,
            "trial_end": "2021-10-23T00:36:24.000Z",
            "users_admin_only": false,
          },
        },
      ],
      "_type": "aquasec_account",
      "additionalCredits": 0,
      "associateTags": false,
      "awsMarketplaceId": "",
      "blockDefaultGroup": false,
      "createdOn": 1633739784000,
      "cspEnabled": true,
      "currentPlan": 57,
      "disableDefaultEmails": false,
      "displayName": "10630",
      "enforceOauth": false,
      "globalSuppressionsAdminOnly": false,
      "id": "10630",
      "integrationsAdminOnly": false,
      "manager": "austin.kelleher@jupiterone.dev",
      "name": "10630",
      "numAllowedKeys": 1,
      "presuppressPlugins": false,
      "rootUser": "austin.kelleher@jupiterone.dev",
      "source": "aquasec",
      "stripeId": undefined,
      "suppressionsAdminOnly": false,
      "trialEnd": 1634949384000,
      "usersAdminOnly": false,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "aquasec_account",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
